### ISSUE TITLE
TabbedPage does not trigger when reselected current tab

### ISSUE THREAD

**Issue by espritm** [dotnet/maui#17877]
### Description

The TabbedPage provide an event called "CurrentPageChanged", which is triggered when user click on Tab1, then on Tab2, etc..
The TabbedPage does not provide event when user click on Tab1, then on Tab1, and on Tab1 again, etc...

On native Android for instance, with a BottomNavigationView (TabbedPage with tabs at bottom), this event is called [OnItemReselectedListener](https://developer.android.com/reference/com/google/android/material/navigation/NavigationBarView.OnItemReselectedListener#summary)

[This documentation](https://developer.android.com/reference/com/google/android/material/navigation/NavigationBarView#setOnItemSelectedListener(com.google.android.material.navigation.NavigationBarView.OnItemSelectedListener)) says "This listener will also be notified when the currently selected item is reselected, unless an NavigationBarView.OnItemReselectedListener has also been set".

I have been digging up a while and [here](https://github.com/dotnet/maui/blob/0afc1c769e46b08530f1747344f70a95c56f8a31/src/Controls/src/Core/Platform/Android/TabbedPageManager.cs#L880C11-L880C11), we do nothing when a item is selected if its ID is the same as the current selected item id.

I tried to get this event by myself using Handlers, but it seems it is not possible - I couldn't find the solution. 
I can't find a way to get the reference to that BottomNavigationBar using handlers, so I don't see workaround.
Here is my sample https://github.com/espritm/Maui.TabbedPage.OnItemReselected

### Steps to Reproduce

1. Open the sample project
2. Build and deploy for Android
3. When clicking on Tab1, then on Tab2, the event "CurrentPageChanged" is triggered

Expected outcome : 
4. When clicking on Tab1, then on Tab1 again, an event "CurrentPageReselected" is triggered.

Actual outcome :
4. When clicking on Tab1, then on Tab1 again, the event "CurrentPageChanged" is not triggered, and we have no other solution.

### Link to public reproduction project repository

https://github.com/espritm/Maui.TabbedPage.OnItemReselected

### Version with bug

7.0.92

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, I was *not* able test on other platforms

### Did you find any workaround?

No, tried to use Handlers but could not access to the appropriate instances of native views.

**Comment by BentAHogset** [comment]
Any progress on this? I'm struggling with the same issue...

**Comment by nord-** [comment]
I'm also interested in the progress of this issue. I would like to navigate to "tab root" if the current tab is selected (again).

**Comment by LeoJHarris** [comment]
👀

**Comment by pictos** [comment]
I can't see how this is a bug... If you're in the same page the `CurrentPageChanged` shouldn't trigger, as it does. I think what you want is a new event or Command that will fire when that happens.

For now, you can implement your own action customizing the Handler

**Comment by LeoJHarris** [comment]
@pictos Agree, seems like more of a feature request then anything else. If it can be handled natively then a handler should work somehow.

**Comment by etm-ubp** [comment]
Would be happy and greatful to have a sample handler to handle this please.
I've not been able to achieve it my self unfortunately.

**Comment by pictos** [comment]
There you go

```cs

using static System.Reflection.BindingFlags;

    // TabbedPage.Loaded event 
	void OnLoaded(object? sender, EventArgs e)
	{
#if ANDROID

		if (Handler is null)
			return;

					        // BindingFlags
		var flags = Instance | NonPublic;


		Type tabbedPageType = typeof(Microsoft.Maui.Controls.TabbedPage);

		// Get the field information for the _tabbedPageManager field
		var tabManager = tabbedPageType.GetField("_tabbedPageManager", flags)?.GetValue(this);

		Debug.Assert(tabManager is not null);

		var tabLayout = tabManager.GetType().GetField("_tabLayout", flags)?.GetValue(tabManager);

		if (tabLayout is Google.Android.Material.Tabs.TabLayout tLayout)
		{
			tLayout.TabReselected += (s, e) =>
			{
				
				var item = e.Tab;
				_ = 1;
			};

			return;
		}

		var bottomNavigation = tabManager.GetType().GetField("_bottomNavigationView", flags)?.GetValue(tabManager);

		if (bottomNavigation is Google.Android.Material.BottomNavigation.BottomNavigationView bNav)
		{
			bNav.ItemReselected += (s, e) =>
			{
				var item = e.Item;
				_ = 1;
			};
		}
#endif
	}
```

@PureWeen, this is a good case to make public or create some extension points around the `TabbedPageManager`

**Comment by HarishKumarSF4517** [comment]
This issue has been verified in Visual Studio Code 1.103.2 with MAUI latest versions 9.0.110. It can be reproduced on the Android platform. And unable to run in iOS using the same sample, resulted in exception. I have attached the snap below:

<img width="824" height="485" alt="Image" src="https://github.com/user-attachments/assets/fb0854d2-784c-4611-a70c-6d8a6ecc7561" />
